clients/src/main/java/org/apache/kafka/server/authorizer/Action.java

```diff
@@ -36,22 +36,22 @@ public Action(AclOperation operation,
                   int resourceReferenceCount,
                   boolean logIfAllowed,
                   boolean logIfDenied) {
-        this.operation = operation;
-        this.resourcePattern = resourcePattern;
+        this.operation = Objects.requireNonNull(operation, "operation can't be null");
+        this.resourcePattern = Objects.requireNonNull(resourcePattern, "resourcePattern can't be null");
         this.logIfAllowed = logIfAllowed;
         this.logIfDenied = logIfDenied;
         this.resourceReferenceCount = resourceReferenceCount;
     }
 
     /**
-     * Resource on which action is being performed.
+     * Resource on which action is being performed. never null
```

Comment:
I would include this in the sentence, "Returns a non-null resource pattern on which this action is being performed".